{"_id":"tryit","_rev":"6-02b3b30c2b08db004160946029c6522a","name":"tryit","description":"Module to wrap try-catch for better performance and cleaner API.","dist-tags":{"latest":"1.0.2"},"versions":{"1.0.0":{"name":"tryit","version":"1.0.0","description":"Module to wrap try-catch for better performance and cleaner API.","main":"tryit.js","repository":{"type":"git","url":"git@github.com:HenrikJoreteg/tryit.git"},"keywords":["errors","try","errorhandling"],"author":{"name":"Henrik Joreteg","email":"henrik@andyet.net"},"license":"MIT","_id":"tryit@1.0.0","dist":{"shasum":"554212b77d30b5aa81fd18003e39832baaba16b8","tarball":"https://registry.npmjs.org/tryit/-/tryit-1.0.0.tgz"},"_npmVersion":"1.1.62","_npmUser":{"name":"henrikjoreteg","email":"henrik@andyet.net"},"maintainers":[{"name":"henrikjoreteg","email":"henrik@andyet.net"}],"directories":{}},"1.0.1":{"name":"tryit","description":"Module to wrap try-catch for better performance and cleaner API.","version":"1.0.1","author":{"name":"Henrik Joreteg","email":"henrik@andyet.net"},"devDependencies":{"tap-spec":"^2.1.2","tape":"^3.0.3"},"keywords":["errors","try","errorhandling"],"license":"MIT","main":"tryit.js","repository":{"type":"git","url":"git@github.com:HenrikJoreteg/tryit.git"},"scripts":{"test":"node test/test.js | tap-spec"},"gitHead":"f948001779cfc9665a9edcd081ee355c74282287","bugs":{"url":"https://github.com/HenrikJoreteg/tryit/issues"},"homepage":"https://github.com/HenrikJoreteg/tryit","_id":"tryit@1.0.1","_shasum":"23a53fffb5f49baf3a00f47abda7ffadecfb85eb","_from":".","_npmVersion":"2.1.5","_nodeVersion":"0.10.30","_npmUser":{"name":"henrikjoreteg","email":"henrik@andyet.net"},"maintainers":[{"name":"henrikjoreteg","email":"henrik@andyet.net"}],"dist":{"shasum":"23a53fffb5f49baf3a00f47abda7ffadecfb85eb","tarball":"https://registry.npmjs.org/tryit/-/tryit-1.0.1.tgz"},"directories":{}},"1.0.2":{"name":"tryit","description":"Module to wrap try-catch for better performance and cleaner API.","version":"1.0.2","author":{"name":"Henrik Joreteg","email":"henrik@andyet.net"},"devDependencies":{"tap-spec":"^2.1.2","tape":"^3.0.3"},"keywords":["errors","try","errorhandling"],"license":"MIT","main":"tryit.js","repository":{"type":"git","url":"git+ssh://git@github.com/HenrikJoreteg/tryit.git"},"scripts":{"test":"node test/test.js | tap-spec"},"gitHead":"b567b4feb491e2ee89addd3d06f66d6ec2217cf5","bugs":{"url":"https://github.com/HenrikJoreteg/tryit/issues"},"homepage":"https://github.com/HenrikJoreteg/tryit#readme","_id":"tryit@1.0.2","_shasum":"c196b0073e6b1c595d93c9c830855b7acc32a453","_from":".","_npmVersion":"3.3.8","_nodeVersion":"4.1.0","_npmUser":{"name":"henrikjoreteg","email":"henrik@joreteg.com"},"dist":{"shasum":"c196b0073e6b1c595d93c9c830855b7acc32a453","tarball":"https://registry.npmjs.org/tryit/-/tryit-1.0.2.tgz"},"maintainers":[{"name":"henrikjoreteg","email":"henrik@andyet.net"}],"directories":{}}},"readme":"# tryit\n\nTiny module wrapping try/catch in JavaScript. \n\nIt's *literally 11 lines of code*, [just read it](tryit.js) that's all the documentation you'll need.\n\n\n## install\n\n```\nnpm install tryit\n```\n\n## usage \n\nWhat you'd normally do:\n```js\ntry {\n    dangerousThing();\n} catch (e) {\n    console.log('something');\n}\n```\n\nWith try-it (all it does is wrap try-catch)\n```js\nvar tryit = require('tryit');\n\ntryit(dangerousThing);\n```\n\nYou can also handle the error by passing a second function\n```js\ntryit(dangerousThing, function (e) {\n    if (e) {\n        console.log('do something');\n    }\n})\n```\n\nThe second function follows error-first pattern common in node. So if you pass a callback it gets called in both cases. But will have an error as the first argument if it fails.\n\n## WHAT? WHY DO THIS!? \n\nPrimary motivation is having a clean way to wrap things that might fail, where I don't care if it fails. I just want to try it. \n\nThis includes stuff like blindly reading/parsing stuff from localStorage in the browser. If it's not there or if parsing it fails, that's fine. But I don't want to leave a bunch of empty `catch (e) {}` blocks in the code.\n\nObviously, this is useful any time you're going to attempt to read some unknown data structure.\n\nIn addition, my understanding is that it's hard for JS engines to optimize code in try blocks. By actually passing the code to be executed into a re-used try block, we can avoid having to have more than a single try block in our app. Again, this is not a primary motivation, just a potential side benefit. \n\n\n## license\n\n[MIT](http://mit.joreteg.com/)\n","maintainers":[{"name":"henrikjoreteg","email":"henrik@andyet.net"}],"time":{"modified":"2015-10-29T16:15:50.649Z","created":"2013-10-24T17:45:00.249Z","1.0.0":"2013-10-24T17:45:01.381Z","1.0.1":"2015-01-05T16:47:06.551Z","1.0.2":"2015-10-29T16:15:50.649Z"},"author":{"name":"Henrik Joreteg","email":"henrik@andyet.net"},"repository":{"type":"git","url":"git+ssh://git@github.com/HenrikJoreteg/tryit.git"},"homepage":"https://github.com/HenrikJoreteg/tryit#readme","keywords":["errors","try","errorhandling"],"bugs":{"url":"https://github.com/HenrikJoreteg/tryit/issues"},"license":"MIT","readmeFilename":"README.md","_attachments":{},"_etag":"\"3LZTOK7CZ9CXR79BUIKJ8XZM2\""}